presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
+                    .collect(toImmutableList());
+            if (!argumentsNotInGroupBy.isEmpty()) {
+                throw new SemanticException(
+                        INVALID_PROCEDURE_ARGUMENTS,
+                        node,
+                        "Arguments in GROUPING() must be in the query's GROUP BY columns. Remove columns '%s' from GROUPING() or add them to the GROUP BY.",
```

Comment:
The following query is legal and executes correctly.
```
SELECT origin_state, sum(origin_zip), grouping(origin_state)
FROM shipping
GROUP BY 1
```

The assertion in the code still works because `AggregationAnalyzer` is constructed using identifiers that have already been resolved so I don't need to do anything with `fieldIndexes`.

That being said the error message can probably be improved. For the following query
```
SELECT origin_state, sum(origin_zip), grouping(destination_state)
FROM shipping
GROUP BY 1
```
this is what Postgres returns `ERROR:  arguments to GROUPING must be grouping expressions of the associated query level`, which is pretty similar to what I'm saying.

How about this: `The arguments to GROUPING must be expressions referenced the GROUP BY at the associated query level`, and I get rid of the second part of the error message about removing columns?